src/Admin/IdentityServer/CustomClaimsPrincipalFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Security.Claims;
+using Bit.Core.Settings;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+
+public class CustomClaimsPrincipalFactory : UserClaimsPrincipalFactory<IdentityUser>
```

Comment:
❓ is this named to be a generic catch all for future custom claims or is this more intended to be a one off for specifically adding the Role custom claim? If it is the latter, renaming it to something like `RoleClaimPrincipalFactory` might make its purpose more clear.

Reply:
As far as I know, it's not possible to implement multiple `UserClaimsPrincipalFactory` at the same time, and I was concerned that giving it a specific name would imply that adding a new custom claim would require adding a new custom `UserClaimsPrincipalFactory`. Since this would be the place future claims would be added, I'm in favor of a generic name.

Reply:
Thank you for the explanation; that makes sense. I didn't know to ask if it was possible or not! Thanks @trmartin4 for the clarification!